avocado/plugins/diff.py

```diff
@@ -0,0 +1,422 @@
+        self.std_diff_output = True
+
+    def configure(self, parser):
+        """
+        Add the subparser for the diff action.
+
+        :param parser: Main test runner parser.
+        """
+        parser = super(Diff, self).configure(parser)
+
+        parser.add_argument("jobids",
+                            default=[], nargs=2,
+                            metavar="<JOB>",
+                            help='A job reference, identified by a (partial) '
+                            'unique ID (SHA1) or test results directory.')
+
+        parser.add_argument('--html', type=str,
+                            metavar='FILE',
+                            help='Enable HTML output to the FILE where the '
+                            'result should be written.')
+
+        parser.add_argument('--open-browser',
+                            action='store_true',
+                            default=False,
+                            help='Generate and open a HTML report in your '
+                            'preferred browser. If no --html file is '
+                            'provided, create a temporary file.')
+
+        parser.add_argument('--diff-filter',
+                            dest='diff_filter',
+                            type=self._validate_filters,
+                            default=['cmdline', 'time', 'variants',
+                                     'results', 'config', 'sysinfo'],
+                            help='Comma separated filter of diff sections: '
+                            '(no)cmdline,(no)time,(no)variants,(no)results,\n'
+                            '(no)config,(no)sysinfo (defaults to all '
+                            'enabled).')
+
+        parser.add_argument('--paginator',
+                            choices=('on', 'off'), default='on',
+                            help='Turn the paginator on/off. '
+                            'Current: %(default)s')
+
+        parser.add_argument('--create-reports', action='store_true',
+                            help='Create temporary files with job reports '
+                            '(to be used by other diff tools)')
+
+        parser.epilog = 'By default, a textual diff report is generated '\
+                        'in the standard output.'
+
+    def run(self, args):
+        job1_dir, job1_id = self._setup_job(args.jobids[0])
+        job2_dir, job2_id = self._setup_job(args.jobids[1])
+
+        job1_data = self._get_job_data(job1_dir)
+        job2_data = self._get_job_data(job2_dir)
+
+        report_header = 'Avocado Job Report\n'
+        job1_results = [report_header]
+        job2_results = [report_header]
+
+        if 'cmdline' in args.diff_filter:
+            cmdline1 = self._get_command_line(job1_dir)
+            cmdline2 = self._get_command_line(job2_dir)
+
+            if str(cmdline1) != str(cmdline2):
+                command_line_header = ['\n',
+                                       '# COMMAND LINE\n']
```

Comment:
Not really nacking. Just strongly discouraging and I'll do my best to try to fix it if this becomes accepted. I know it's not rst format, but it's a section name so it should be visible. IMO it should be visible the most from the diff as it divides the sections.

PS: It's not only about color, because it should be visible even when using bw.